sympy/tensor/array/ndim_array.py

```diff
@@ -129,34 +129,40 @@ def f(pointer):
     @classmethod
     def _handle_ndarray_creation_inputs(cls, iterable=None, shape=None, **kwargs):
         from sympy.matrices.matrices import MatrixBase
-
-        if shape is None and iterable is None:
-            shape = ()
-            iterable = ()
-        # Construction from another `NDimArray`:
-        elif shape is None and isinstance(iterable, NDimArray):
-            shape = iterable.shape
-            iterable = list(iterable)
-        # Construct N-dim array from an iterable (numpy arrays included):
-        elif shape is None and isinstance(iterable, Iterable):
-            iterable, shape = cls._scan_iterable_shape(iterable)
-
-        # Construct N-dim array from a Matrix:
-        elif shape is None and isinstance(iterable, MatrixBase):
-            shape = iterable.shape
-
-        # Construct N-dim array from another N-dim array:
-        elif shape is None and isinstance(iterable, NDimArray):
-            shape = iterable.shape
+        from sympy.tensor.array import SparseNDimArray
+        from sympy import Dict
+
+        if shape is None:
+            if iterable is None:
```

Comment:
is this change necessary?

Reply:
This change is to refactor the code so that ```shape is None``` will not be repeated several times

Reply:
OK, open a new PR next time. A separate PR on this issue would already be mergeable.